src/System.CommandLine/CommandLineConfiguration.cs

```diff
@@ -11,12 +11,28 @@
 
 namespace System.CommandLine
 {
+    /// <summary>
+    /// Represents the configuration used by the <see cref="Parser"/>.
+    /// </summary>
     public class CommandLineConfiguration
     {
         private readonly IReadOnlyCollection<InvocationMiddleware> _middlewarePipeline;
         private readonly Func<BindingContext, IHelpBuilder> _helpBuilderFactory;
         private readonly SymbolSet _symbols = new SymbolSet();
 
+        /// <summary>
+        /// Initializes a new instance of the CommandLineConfiguration class.
+        /// </summary>
+        /// <param name="symbols">The symbols to parse.</param>
+        /// <param name="argumentDelimiters">The characters used to delimit arguments.</param>
+        /// <param name="enablePosixBundling"><c>true</c> to enable POSIX bundling; otherwise, <c>false</c>.</param>
+        /// <param name="enableDirectives"><c>true</c> to enable directive parsing; otherwise, <c>false</c>.</param>
```

Comment:
For directives and POSIX bundling, more context or an example would probably be useful.